dropwizard-core/src/test/java/io/dropwizard/cli/CommandTest.java

```diff
@@ -0,0 +1,82 @@
+package io.dropwizard.cli;
+
+import com.google.common.base.Optional;
+import io.dropwizard.Application;
+import io.dropwizard.Configuration;
+import io.dropwizard.setup.Bootstrap;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.util.JarLocation;
+import net.sourceforge.argparse4j.inf.Namespace;
+import net.sourceforge.argparse4j.inf.Subparser;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CommandTest {
+    private static class TestCommand extends Command {
+        protected TestCommand() {
+            super("test", "test");
+        }
+
+        @Override
+        public void configure(Subparser subparser) {
+            subparser.addArgument("types").choices("a", "b", "c").help("Type to use");
+        }
+
+        @Override
+        public void run(Bootstrap<?> bootstrap, Namespace namespace) throws Exception {
+        }
+    }
+
+    private final Application<Configuration> app = new Application<Configuration>() {
+        @Override
+        public void run(Configuration configuration, Environment environment) throws Exception {
+        }
+    };
+
+    private final JarLocation location = mock(JarLocation.class);
+    private final Bootstrap<Configuration> bootstrap = new Bootstrap<>(app);
+    private final ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    private final ByteArrayOutputStream stdErr = new ByteArrayOutputStream();
+    private final Command command = new TestCommand();
+    private Cli cli;
+
+    @Before
+    @SuppressWarnings("unchecked")
```

Comment:
Do we need this annotation? I believe all operations are type-safe in the method.